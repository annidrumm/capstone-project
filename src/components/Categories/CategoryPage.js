import React from 'react'
import styled from 'styled-components/macro'
import forkIcon from '../../icons/fork.svg'
import gymIcon from '../../icons/gym.svg'
import beautyIcon from '../../icons/beauty.svg'
import fashionIcon from '../../icons/mannequin.svg'
import InteriorIcon from '../../icons/vase.svg'
import travelIcon from '../../icons/plane.svg'

export default function CategoryPage() {
  return (
    <ContainerStyled>
      <HeadlineStyled>
        What category are<br></br> you looking for?
      </HeadlineStyled>
      <section>
        <ButtonStyled category="food">
          <IconsStyled src={forkIcon} alt="" />
          <ButtonTextStyled>Food</ButtonTextStyled>
        </ButtonStyled>
        <ButtonStyled category="fitness">
          <IconsStyled src={gymIcon} alt="" />
          <ButtonTextStyled>Fitness</ButtonTextStyled>
        </ButtonStyled>
      </section>
      <section>
        <ButtonStyled category="beauty">
          <img src={beautyIcon} alt="" />
          <ButtonTextStyled>Beauty</ButtonTextStyled>
        </ButtonStyled>
        <ButtonStyled category="fashion">
          <img src={fashionIcon} alt="" />
          <ButtonTextStyled>Fashion</ButtonTextStyled>
        </ButtonStyled>
      </section>
      <section>
        <ButtonStyled category="interior">
          <img src={InteriorIcon} alt="" />
          <ButtonTextStyled>Interior</ButtonTextStyled>
        </ButtonStyled>
        <ButtonStyled category="travel">
          <img src={travelIcon} alt="" />
          <ButtonTextStyled>Travel</ButtonTextStyled>
        </ButtonStyled>
      </section>
    </ContainerStyled>
  )
}

const ContainerStyled = styled.section`
  text-align: center;
`

const HeadlineStyled = styled.h1`
  font: var(--font-weight-regular);
  font-size: 35px;
`
const ButtonStyled = styled.button`
  width: 128px;
  height: 101px;
  background: transparent;
  margin: 10px;
  border-radius: 10%;

  border-color: ${({ category }) =>
    category === 'food'
      ? 'var(--darkyellow)'
      : category === 'fitness'
      ? 'var(--lightpink)'
      : category === 'beauty'
      ? 'var(--lightgreen)'
      : category === 'fashion'
      ? 'var(--lightblue)'
      : category === 'interior'
      ? 'var(--lightpurple)'
      : 'var(--darkyellow)'};
`
const IconsStyled = styled.img`
  margin-top: 15px;
  vertical-align: top;
`
const ButtonTextStyled = styled.h2`
  color: var(--font-color-white);
  font-size: 20px;
  font-weight: var(--font-weight-light);
  vertical-align: bottom;
  margin-bottom: 7px;
  margin-top: 3px;
`
