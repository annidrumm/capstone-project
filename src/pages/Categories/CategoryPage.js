import React from 'react'
import styled from 'styled-components/macro'
import Headline from '../../components/Headline/Headline'
import CategoryButton from '../../components/CategoryButton/CategoryButton'

export default function CategoryPage() {
  return (
    <ContainerStyled>
      <Headline headline="What categories are you looking for?" />
      <CategoryButton />
    </ContainerStyled>
  )
}

const ContainerStyled = styled.section`
  text-align: center;
`
