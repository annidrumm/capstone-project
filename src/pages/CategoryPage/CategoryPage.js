import React from 'react'
import styled from 'styled-components/macro'
import CategoryButtons from '../../components/CategoryButtons/CategoryButtons'
import HeaderCategoryPage from '../../components/Header/HeaderCategoryPage'

export default function CategoryPage() {
  return (
    <ContainerStyled>
      <HeaderCategoryPage headline="Categories" />
      <CategoryButtons />
    </ContainerStyled>
  )
}

const ContainerStyled = styled.section`
  text-align: center;
  margin: 30px;
`
