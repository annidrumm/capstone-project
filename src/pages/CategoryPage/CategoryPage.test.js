import React from 'react'
import renderer from 'react-test-renderer'
import CategoryPage from './CategoryPage'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

describe('CategoryPage', () => {
  const headline = 'What categories are you looking for?'

  it('display headline', () => {
    const { getByText } = render(<CategoryPage headline={headline} />)
    expect(getByText(headline)).toBeInTheDocument()
  })
  it('renders CategoryPage correctly', () => {
    const tree = renderer.create(<CategoryPage />)
    expect(tree).toMatchSnapshot()
  })
})
