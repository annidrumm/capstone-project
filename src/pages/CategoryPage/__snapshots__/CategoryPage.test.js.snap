// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryPage renders CategoryPage correctly 1`] = `
<section
  className="CategoryPage__ContainerStyled-inp28z-0 brhHgo"
>
  <section
    className="sc-AxjAm ljJdnD"
  >
    <svg
      className="sc-AxiKw iRsRKy"
      onClick={[Function]}
    >
      back.svg
    </svg>
    <h2
      className="sc-AxirZ iJYoHj"
    >
      Categories
    </h2>
  </section>
  <section
    className="CategoryButtons__SectionStyled-fe33yf-1 eUTwsm"
  >
    <section
      className="CategoryButtons__ContainerStyled-fe33yf-0 ggAkgj"
    >
      <section>
        <a
          href="/food"
          onClick={[Function]}
        >
          <button
            className="CategoryButtons__ButtonStyled-fe33yf-2 eEhnNm"
          >
            <img
              alt="fork and spoon"
              className="CategoryButtons__IconsStyled-fe33yf-3 gHSayS"
              src="fork.svg"
            />
            <h2
              className="CategoryButtons__H2Styled-fe33yf-4 fWGdzg"
            >
              Food
            </h2>
          </button>
        </a>
        <a
          href="/fitness"
          onClick={[Function]}
        >
          <button
            className="CategoryButtons__ButtonStyled-fe33yf-2 jLejSs"
          >
            <img
              alt="dumbbell"
              className="CategoryButtons__IconsStyled-fe33yf-3 gHSayS"
              src="gym.svg"
            />
            <h2
              className="CategoryButtons__H2Styled-fe33yf-4 fWGdzg"
            >
              Fitness
            </h2>
          </button>
        </a>
      </section>
      <section>
        <a
          href="/beauty"
          onClick={[Function]}
        >
          <button
            className="CategoryButtons__ButtonStyled-fe33yf-2 YSSmf"
          >
            <img
              alt="lipsticks"
              src="beauty.svg"
            />
            <h2
              className="CategoryButtons__H2Styled-fe33yf-4 fWGdzg"
            >
              Beauty
            </h2>
          </button>
        </a>
        <a
          href="/fashion"
          onClick={[Function]}
        >
          <button
            className="CategoryButtons__ButtonStyled-fe33yf-2 jVhgki"
          >
            <img
              alt="fashion mannequin"
              src="mannequin.svg"
            />
            <h2
              className="CategoryButtons__H2Styled-fe33yf-4 fWGdzg"
            >
              Fashion
            </h2>
          </button>
        </a>
      </section>
      <section>
        <a
          href="/interior"
          onClick={[Function]}
        >
          <button
            className="CategoryButtons__ButtonStyled-fe33yf-2 jCiqwi"
          >
            <img
              alt="vase"
              src="vase.svg"
            />
            <h2
              className="CategoryButtons__H2Styled-fe33yf-4 fWGdzg"
            >
              Interior
            </h2>
          </button>
        </a>
        <a
          href="/travel"
          onClick={[Function]}
        >
          <button
            className="CategoryButtons__ButtonStyled-fe33yf-2 eEhnNm"
          >
            <img
              alt="plane"
              src="plane.svg"
            />
            <h2
              className="CategoryButtons__H2Styled-fe33yf-4 fWGdzg"
            >
              Travel
            </h2>
          </button>
        </a>
      </section>
    </section>
  </section>
</section>
`;
